src/py21cmfast/src/ps.c

```diff
@@ -2256,8 +2389,69 @@ float GaussLegendreQuad_Nion(int Type, int n, float growthf, float M2, float sig
         .LimitMass_Fesc = Mlim_Fesc
     };
 
-    if(delta2 > delta1){
-        return 1.;
+    if (FAST_FCOLL_TABLES){ //JBM: Fast tables. Assume sharp Mturn, not exponential cutoff.
+
+      double LogMass=log(MassTurnover);
+      int MassBin = (int)floor( (LogMass - MinMass )*inv_mass_bin_width );
+      double MassBinLow = MinMass + mass_bin_width*(double)MassBin;
+      double sigmaM1 = Sigma_InterpTable[MassBin] + ( LogMass - MassBinLow )*( Sigma_InterpTable[MassBin+1] - Sigma_InterpTable[MassBin] )*inv_mass_bin_width;
+      nu_lower_limit = (delta1-delta2)*(delta1-delta2)/(sigmaM1*sigmaM1-sigma2*sigma2)/growthf/growthf;
+
+
+      LogMass = log(1.5e9); //jbm could be done outside and it'd be even faster
+      int MassBinpivot = (int)floor( (LogMass - MinMass )*inv_mass_bin_width );
+      double MassBinLowpivot = MinMass + mass_bin_width*(double)MassBinpivot;
+      double sigmapivot1 = Sigma_InterpTable[MassBinpivot] + ( LogMass - MassBinLowpivot )*( Sigma_InterpTable[MassBinpivot+1] - Sigma_InterpTable[MassBinpivot] )*inv_mass_bin_width;
+      double nupivot1 = (delta1-delta2)*(delta1-delta2)/(sigmapivot1*sigmapivot1)/growthf/growthf; //note, it does not have the sigma2 on purpose.
+
+      LogMass = log(5.3e5); //jbm could be done outside and it'd be even faster
+      MassBinpivot = (int)floor( (LogMass - MinMass )*inv_mass_bin_width );
+      MassBinLowpivot = MinMass + mass_bin_width*(double)MassBinpivot;
+      double sigmapivot2 = Sigma_InterpTable[MassBinpivot] + ( LogMass - MassBinLowpivot )*( Sigma_InterpTable[MassBinpivot+1] - Sigma_InterpTable[MassBinpivot] )*inv_mass_bin_width;
+      double nupivot2 = (delta1-delta2)*(delta1-delta2)/(sigmapivot2*sigmapivot2)/growthf/growthf;
+
+
+      double beta1 = (Alpha_star+Alpha_esc) * 9.0 * (0.5); //exponent for Fcollapprox for nu>nupivot1 (large M)
+      double beta2 = (Alpha_star+Alpha_esc) * 13.6 * (0.5); //exponent for Fcollapprox for nupivot2<nu<nupivot1 (small M)
+      double beta3 = (Alpha_star+Alpha_esc) * 21.0 * (0.5); //exponent for Fcollapprox for nu<nupivot2 (smallest M)
+    //beta2 fixed by continuity.
+
+
+      double corrfact=sigma2/(delta1-delta2)*growthf+1e-10; //small number added to avoid infinities
+      double nucrit_sigma2 = 1.0/corrfact/corrfact; //above this nu sigma2>sigma1, so HMF=0
+
```

Comment:
Could re-write as;

corrfact = sigma2 * pow( (delta1 - delta2)*growth , -1.);
nucrit_sigma2 = pow( corrfact, -2.);

Generally speaking multiplication is computationally faster than division, so often I err on making things multiplications. I think from memory, calling pow for inverse powers is also faster than normal division (although the difference is likely to be extremely minimal).

Reply:
Good point, I've fixed this too to do fewer operations, and also to be just one line since we don't need corrfact anywhere else.